Sources/Sentry/SentryProfiler.mm

```diff
@@ -212,9 +212,16 @@ - (void)stopForReason:(SentryProfilerTruncationReason)reason
     }
 
 #    if SENTRY_HAS_UIKIT
-    // if SentryOptions.enableAutoPerformanceTracing is NO, then we need to stop the frames tracker
-    // from running outside of profiles because it isn't needed for anything else
-    if (![[[[SentrySDK currentHub] getClient] options] enableAutoPerformanceTracing]) {
+    // if SentryOptions.enableAutoPerformanceTracing is NO and appHangsV2Disabled, which uses the
+    // frames tracker, then we need to stop the frames tracker from running outside of profiles
+    // because it isn't needed for anything else
```

Comment:
`m`: I think there's something missing.

```suggestion
    // if SentryOptions.enableAutoPerformanceTracing is NO and appHangsV2Disabled, which uses the
    // frames tracker, is YES, then we need to stop the frames tracker from running outside of profiles
    // because it isn't needed for anything else
```